### PR TITLE
feat: Attachments support

### LINKED ISSUE #574: Support for attachments in go sdk
### Problem Statement

It will be great if you could add the support for attachments for go SDK. 

### Solution Brainstorm

_No response_

### PR DESCRIPTION
**Pull request by edigaryev**
This is similar to how [`sentry-rust` supports attachments](https://docs.rs/sentry/0.31.5/sentry/struct.Scope.html#method.add_attachment).

Resolves #574.

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -94,6 +94,38 @@ func getRequestBodyFromEvent(event *Event) []byte {
 	return nil
 }
 
+func encodeAttachment(enc *json.Encoder, b io.Writer, attachment *Attachment) error {
```
Comment (tonyo) [inline transport.go:97]: Please add a test in `transport_test.go` that tests this function.

**Review by tonyo — COMMENTED** [review]
Looking good, just one ask about test coverage 👍

File: transport.go (hunk 1 above)
Comment (edigaryev) [reply transport.go:97]: PTAL at https://github.com/getsentry/sentry-go/pull/670/commits/52caef25b90d3055ea9ee5066536c2493a4d4fed.

Not sure if testing `encodeAttachment()` makes sense in the presence of tests like `TestEnvelopeFromErrorBody`, which already will call that function under the hood if we set `event.attachments` to non-`nil`.

Comment (tonyo) [reply transport.go:97]: Yep, what you added is actually what I had in mind 👍

**Review by tonyo — APPROVED** [review]

**Comment by tonyo** [comment]
Thanks @edigaryev 🎉

**Comment by tonyo** [comment]
The attachments support is now released as part of https://github.com/getsentry/sentry-go/releases/tag/v0.23.0 👍

**Pull request merged**